packages/node-http-handler/src/node-http-handler.ts

```diff
@@ -42,8 +42,9 @@ export class NodeHttpHandler implements HttpHandler {
     this.connectionTimeout = connectionTimeout;
     this.socketTimeout = socketTimeout;
     const keepAlive = true;
-    this.httpAgent = httpAgent || new hAgent({ keepAlive });
-    this.httpsAgent = httpsAgent || new hsAgent({ keepAlive });
+    const maxSockets = 50;
```

Comment:
Would it make sense to allow passing this option alongside other options in the constructor `{..., maxSockets = 50 }`? AWS.config.update used to allow passing maxSockets if I remembered correctly, and this would help with high-traffic calls for slow API endpoints

It'd make migration from v2 easier, but it could be argued to be redundant if httpsAgent is supported

Reply:
Both the values for `keepAlive` and `maxSockets` can be set by user by passing a custom user agent as follows:

```js
const { DynamoDBClient } = require("@aws-sdk/client-dynamodb");
const { NodeHttpHandler } = require("@aws-sdk/node-http-handler");
var https = require("https");

var dynamodbClient = new DynamoDBClient({
  requestHandler: new NodeHttpHandler({
    httpsAgent: new https.Agent({
      keepAlive: false,
      maxSockets: 25,
    }),
  }),
});
```

This is documented in Developer Guide in [Configuring maxSockets in Node.js](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/node-configuring-maxsockets.html)

Is this what you're looking for @TysonAndre ?

Reply:
>  but it could be argued to be redundant if httpsAgent is supported

I was already aware of that option - this was more of a question of whether the sdk itself should support that (e.g. for applications that used to pass a generic config object to AWS.config.update() and are now passing that to v3 constructors

Reply:
The modular AWS SDK for JavaScript (v3) doesn't support global configuration as documented in README: https://github.com/aws/aws-sdk-js-v3/#configuration